File: NetFramework/C1.Win.Document/VB/SsrsViewer/CredentialDialog.vb
Imports System.Windows.Forms
Imports System.Net

Public Class CredentialDialog

    Public Function Login(ByVal owner As IWin32Window, ByVal path As String, ByVal nc As NetworkCredential) As NetworkCredential

        tbPath.Text = path
        If nc IsNot Nothing Then
            tbUser.Text = nc.UserName
            tbPassword.Text = nc.Password
            tbDomain.Text = nc.Domain
        End If
        ActiveControl = tbUser
        If ShowDialog(owner) <> System.Windows.Forms.DialogResult.OK Then
            Return Nothing
        End If
        Return New NetworkCredential(tbUser.Text, tbPassword.Text, tbDomain.Text)

    End Function

    Public Shared Function DoLogin(ByVal owner As IWin32Window, ByVal path As String, ByVal nc As NetworkCredential) As NetworkCredential

        Using f As New CredentialDialog()
            Return f.Login(owner, path, nc)
        End Using

    End Function

End Class

File: NetFramework/WinForms/CS/ControlExplorerOld/SSRS/CredentialDialog.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Net;

namespace ControlExplorer.SSRS
{
    public partial class CredentialDialog : Form
    {
        public CredentialDialog()
        {
            InitializeComponent();
        }

        public NetworkCredential Login(IWin32Window owner, string path, NetworkCredential nc)
        {
            tbPath.Text = path;
            if (nc != null)
            {
                tbUser.Text = nc.UserName;
                tbPassword.Text = nc.Password;
                tbDomain.Text = nc.Domain;
            }
            ActiveControl = tbUser;
            if (ShowDialog(owner) != System.Windows.Forms.DialogResult.OK)
                return null;
            return new NetworkCredential(tbUser.Text, tbPassword.Text, tbDomain.Text);
        }

        public static NetworkCredential DoLogin(IWin32Window owner, string path, NetworkCredential nc)
        {
            using (CredentialDialog f = new CredentialDialog())
                return f.Login(owner, path, nc);
        }
    }
}
